src/handlers/opportunitiesHandlers.ts

```diff
@@ -38,31 +39,44 @@ const getOpportunities = (req: Request, res: Response): void => {
     });
 };
 
-const getOpportunity = (req: Request, res: Response): void => {
-  const isInteger = (it: string): boolean => !Number.isNaN(parseInt(it, 10))
-    && parseInt(it, 10) % 1 === 0;
-
-  if (!isInteger(req.params.id)) {
+interface GetOpportunityParams {
+  id: number;
+}
+const isGetOpportunityParams = ajv.compile<GetOpportunityParams>({
+  type: 'object',
+  properties: {
+    id: {
+      type: 'integer',
+      minimum: 1,
+    },
+  },
+  required: [
+    'id',
+  ],
+});
+const getOpportunity = (req: Request<GetOpportunityParams>, res: Response): void => {
```

Comment:
While I agree that defining a little `isInteger` function is awkward (there should be an existing easier way), I don't think using AJV to validate a single integer (not JSON) is any less awkward.

Reply:
Worth noting that using a validation library is a common approach to pair with express (and we already have ajv so I didn't wanna lean into joi).  Here's a good writeup if it's useful: https://simonplend.com/how-to-handle-request-validation-in-your-express-api/

The complexity of what we're validating isn't defined by the individual instance but rather the world of possibilities across all endpoints (we want validation errors to endpoint params to be consistent regardless of how complicated the endpoint input is).

Basically though Idon't see ajv as awkward, there are a few functional benefits of using it:

1. consistent error message format across all endpoints
2. ajv is tested and maintained by a large community, so far less risk that we have a bug in our validation / we don't have to test the validators themselves.